### PR TITLE
Test filtering out duplicate ancestors of main parent block for replay

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
## Overview
See ticket

### JIRA ticket:
https://rchain.atlassian.net/browse/RHOL-1122

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: casper/src/test/scala/coop/rchain/casper/util/rholang/InterpreterUtilTest.scala [hunk 1]
```diff
@@ -830,4 +830,61 @@ class InterpreterUtilTest
...
+            b3      <- createBlock[Task](Seq(b1.blockHash, b2.blockHash), deploys = b3DeploysWithCost)
+            _       <- step(0, genesis)
+            _       <- step(1, genesis)
+            _       <- step(2, genesis)
+            dag     <- blockDagStorage.getRepresentation
+            blockHashes <- InterpreterUtil.findMultiParentsBlockHashesForReplay(
+                            Seq(b1, b2),
+                            dag
+                          )
+            _ = withClue("Main parent hasn't been filtered out: ") { blockHashes.size shouldBe (1) }
```
Comment (KentShikama) [inline casper/src/test/scala/coop/rchain/casper/util/rholang/InterpreterUtilTest.scala]: hasn't or has?

File: casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala [hunk 2]
```diff
@@ -223,22 +245,7 @@ object InterpreterUtil {
...
-        case Left((_, status)) =>
-          val parentHashes =
-            parents.map(p => Base16.encode(p.blockHash.toByteArray).take(8))
-          Left(
-            new Exception(
-              s"Failed status while computing post state of $parentHashes: $status"
-            )
-          )
-      }
+    } yield (blockHashesToApply)
```
Comment (KentShikama) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala]: unnecessary parens

File: casper/src/test/scala/coop/rchain/casper/util/rholang/InterpreterUtilTest.scala (hunk 1 above)
Comment (lukasz-golebiewski) [reply casper/src/test/scala/coop/rchain/casper/util/rholang/InterpreterUtilTest.scala]: This is a clue. It gets printed out when the assertion fails. So "hasn't"

**Pull request merged**